src/Types.re

```diff
@@ -1,7 +1,29 @@
+/* https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_InitiateAuth.html#API_InitiateAuth_RequestSyntax */
 type authenticationFlowType =
   | USER_PASSWORD_AUTH
   | USER_SRP_AUTH
   | CUSTOM_AUTH;
+/* all operations found in https://github.com/aws-amplify/amplify-js/blob/master/packages/amazon-cognito-identity-js/src/CognitoUserPool.js except `SignUp` which is in `CognitoUserPool.js`*/
+type operation =
```

Comment:
Those docs I liked above have more operations in them we might want to support down the line.  No need to add them now.  Maybe we should start collecting a references library in the README.

Reply:
I looked at some of those operations. Some of them just take unit and no params so might have to make the params on the client request type optional. I am thinking out loud here. I am not getting how to call some of these operations. For example, if we want to call `getCurrurentContextData` found here:https://github.com/aws-amplify/amplify-js/blob/2fbfde04092c2b0bfddc3d9698975ba7e9c5e572/packages/amazon-cognito-identity-js/src/CognitoUserPool.js#L156

I tried this:
```reason
let getUserContextData = (config, ~username: string, ()) => {

  let params = Js.Dict.empty();
  Js.Dict.set(params, "USERNAME", Js.Json.string(username));
  Client.request(config, InitiateAuth, params)
  ->Future.flatMapOk(res =>
      switch (res.status) {
      | Success(_) =>
        Js.log2("RES", res);
        Future.make(resolve =>
          resolve(
            Belt.Result.Ok(
              {
                res;
              },
            ),
          )
        );

      | Informational(_)
      | Redirect(_)
      | ClientError(_)
      | ServerError(_) =>

        let isErrorResponse = makeResponse(res.json);
        let err = isErrorResponse->__typeGet;
        let msg = isErrorResponse->messageGet;
        let err =
          switch (err) {
          | "InvalidParameterException" => `CognitoInvalidParameter(msg)
          | "NotAuthorizedException" => `CognitoNotAuthorized(msg)
          | _ => `CognitoUnknownError(msg)
          };
        Future.make(resolve => resolve(Belt.Result.Error(err)));
      }
    );
};
```

Then call with:

```reason
// Scratch.re
let getUserContextData =
  auth->Cognito.getUserContextData(~username="sometestuser2");
```

The function runs and does nothing. What does that mean? Also, the function takes no params but obviously it must because it needs a username. How would I read the signature in the linked function in order to call it? What operation does that function expect to receive? Teach, brother, teach.